frontend/src/metabase/query_builder/components/CompareAggregations/components/ColumnPicker/ColumnPicker.tsx

```diff
@@ -94,3 +82,62 @@ const Item = forwardRef<
     </div>
   );
 });
+
+function getColumnOptions(comparisonType: string): ItemType[] {
+  if (comparisonType === "offset") {
+    return [
+      {
+        example: "1826, 3004",
+        label: t`Previous value`,
+        value: "offset",
+      },
+      {
+        example: "+2.3%, -0.1%",
+        label: t`Percentage difference`,
+        value: "percent-diff-offset",
+      },
+      {
+        example: "+42, -3",
+        label: t`Value difference`,
+        value: "diff-offset",
+      },
+    ];
+  }
+  if (comparisonType === "moving-average") {
+    return [
+      {
+        example: "1826, 3004",
+        label: t`Moving average value`,
+        value: "moving-average",
+      },
+      {
+        example: "+2.3%, -0.1%",
+        label: t`Percentage difference with moving average`,
+        value: "percent-diff-moving-average",
+      },
+      {
+        example: "+42, -3",
+        label: t`Value difference with moving average`,
+        value: "diff-moving-average",
+      },
+    ];
+  }
+  return [];
+}
+
+function convertValues(
+  values: ColumnType[],
+  comparisonType: string,
+): ColumnType[] {
+  if (comparisonType === "offset") {
+    return values.map(
+      value => value.replace("moving-average", "offset") as ColumnType,
```

Comment:
Issues here won't be caught with typescript. Can we switch to a `Record<ColumnType, ColumnType>` or something like this to make sure it's type-safe?